### PR TITLE
Update the IsGKEAutopilotCluster to understand the new autopilot webhook

### PR DESCRIPTION
**Pull request by nan-yu**
We use the policycontrollerv2.config.common-webhooks.networking.gke.io webhook to detect if a cluster is an Autopilot cluster. However, the webhook was renamed to gkepolicy.config.common-webhooks.networking.gke.io`.
This commit updates the function to return true if the new webhook is found.

b/243938243

### CODE REVIEW

File: pkg/util/autopilot.go [hunk 1]
```diff
@@ -49,6 +49,11 @@ var AutopilotManagedKinds = []schema.GroupVersionKind{
 	admissionregistrationv1.SchemeGroupVersion.WithKind("MutatingWebhookConfigurationList"),
 }
 
+var autopilotWebhooks = []string{
+	"policycontrollerv2.config.common-webhooks.networking.gke.io",
+	"gkepolicy.config.common-webhooks.networking.gke.io",
```
Comment (haiyanmeng) [inline pkg/util/autopilot.go:54]: Please add some comments here: which one is new and when it is introduced, which one is legacy.

File: pkg/util/autopilot.go [hunk 2]
```diff
@@ -76,15 +81,17 @@ func IsGKEAutopilotCluster(c client.Client) (bool, error) {
...
-	} else if apierrors.IsNotFound(err) {
-		return false, nodesErr
+	for _, webhookName := range autopilotWebhooks {
+		webhook := &admissionregistrationv1.ValidatingWebhookConfiguration{}
+		objectKey := client.ObjectKey{Name: webhookName}
+		err := c.Get(context.Background(), objectKey, webhook)
+		if err == nil {
+			return true, nil
+		} else if !apierrors.IsNotFound(err) {
+			return false, err
```
Comment (janetkuo) [inline pkg/util/autopilot.go:91]: The range order would affect the result, if only some of the webhooks are successfully got. We should make the behavior more deterministic.

**Review by janetkuo — APPROVED** [review]
/lgtm

File: pkg/util/autopilot.go (hunk 1 above)
Comment (nan-yu) [reply pkg/util/autopilot.go]: Added a comment about which one is new and which is legacy. I'm still confirming with the autopilot team to see which version was this introduced (waiting for the response). Will add the start version when I get an answer.

File: pkg/util/autopilot.go (hunk 2 above)
Comment (nan-yu) [reply pkg/util/autopilot.go]: Updated with a multierror to collect the apiserver errors.

**Pull request merged**